src/Compilers/Core/Portable/Compilation/ParseOptions.cs

```diff
@@ -69,6 +69,12 @@ public ParseOptions WithKind(SourceCodeKind kind)
             return CommonWithKind(kind);
         }
 
+        /// <summary>
+        /// Get the ParseOptions with latest language version
+        /// </summary>
+        /// <returns></returns>
+        public abstract ParseOptions WithLatestLanguageVersion();
```

Comment:
@dotnet/roslyn-compiler I intended for this to be `internal`. Please let us know how to define such a method for use in the VS layer without a new public API.

Reply:
Why not implement this API in one of the language services? (`ISyntaxTreeFactoryService` seems reasonable place)

If that doesn't work for some reason, you can try making this `internal`, but I'm not sure there is `InternalsVisibleTo` from the compiler to this part of the IDE code. 
If there is no IVT, there is probably a reason why it's not there, and so I would suspect it shouldn't be added.

Reply:
Originally, we couldn't even have IVT between compiler and IDE for legal reasons. Today, we simply don't want it because the compiler API should be sufficient for consumer's needs.

Reply:
If there is concern about a new API why not just make it an extension method? That doesn't need any change in the compiler.

Reply:
Added to the ISyntaxTreeFactoryService.
I am not sure how to add the override extension method for ParseOptions, VisualBasicParseOptions and CSharpParseOptions, so ISyntaxTreeFactoryService looks like a good place